packages/server/src/unstable-core-do-not-import/stream/sse.ts

```diff
@@ -223,6 +240,31 @@ interface ConsumerConfig {
   EventSource: EventSourceLike.AnyConstructor;
 }
 
+async function withTimeout<T>(opts: {
```

Comment:
Would this be better as its own link maybe? timeoutLink feels like a useful primitive and might simplify things?

Reply:
Since it seems to be an issue with all browsers I kinda want it built-in

It's enough to do `kill -9` on a server for `EventSource` to hang in Chrome

Reply:
Fair, MR looks great, happy to approve whenever you say its ready